app/controllers/employee/reach_areas_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Employee::ReachAreasController < ApplicationController
+  before_action :authenticate_employee!, only: %i[new create]
+  before_action :set_shipping_co, only: %i[new create]
+  def new
+    @reach_area = @shipping_co.reach_areas.new
+  end
+
+  def create
+    reach_area = @shipping_co.reach_areas.create(reach_area_params)
+    return redirect_to [:employee, current_employee.shipping_co], alert: t('.duplicate_found') unless reach_area.valid?
```

Comment:
```suggestion
    return redirect_to [:employee, current_employee.shipping_co], alert: t('.duplicate_found') unless reach_area.persisted?
```

Acho que aqui poderia usar um método com mais peso pra verificar o status no banco

Reply:
Boa! Alterado, vlw :)